### PR TITLE
Add LUTTools.swap{MultipleLutPins,LutPinsFromPIPs}()

### PR DESCRIPTION
**Pull request by eddieh-xlnx**
From a logical (not timing) perspective, all of a LUT's input pins can be considered equivalent since its internal mask can be permuted to reflect any pin ordering. Add two helper methods to ease the update of all necessary design state when this operation is undertaken.

* `LUTTools.swapMultipleLutPins()` to take a mapping of old `SitePinInst` to new `SitePin`
* `LUTTools.swapSingleLutPins()` is what was previously `SATRouter.processPinSwaps()` for processing one pin swapping on a single LUT
* `LUTTools.swapLutPinSwapFromPIPs()` to take a `Design` which gets analyzed to determine -- based on its routing PIPs -- which LUT pins need swapping

Note that these methods make no effort to check the validity of the requested swaps -- one could (but shouldn't) request an illegal swap onto another LUT, accidentally or not, or one could try and swap onto the `A6` pin even though the `A5LUT` is being used -- and likely cause the method to return normally.

Two current limitations:
* Ground net (`GLOBAL_LOGIC0`) is not analyzed for LUT pin swaps -- this is not a big deal for RWRoute since `GlobalSignalRouting` does not do any LUT pin swapping.
* Pins on up to two LUTs (assumed to be placed on `x6LUT` and `x5LUT`) can have their pins swapped. This is only not true for `H6LUT.A[1-6]` pins on `SLICEM`s, which also serve as the write address `WA[1-6]` for all LUTs in that slice.

### CODE REVIEW

**Comment by eddieh-xlnx** [comment]
Another thought: should we move `SATRouter.processPinSwaps()` into `LUTTools` too?

**Comment by clavin-xlnx** [comment]
> Another thought: should we move `SATRouter.processPinSwaps()` into `LUTTools` too?

Makes sense to me.

File: src/com/xilinx/rapidwright/router/SATRouter.java [hunk 1]
```diff
@@ -659,7 +659,7 @@ else if (line.startsWith("\"") && line.endsWith("\":")) {
 
         // Make all pin swaps per LUT site simultaneously
         for (Entry<String,HashMap<String,PinSwap>> e : pinSwaps.entrySet()) {
-            processPinSwaps(e.getKey(),new ArrayList<>(e.getValue().values()));
+            processPinSwaps(e.getKey(), e.getValue().values());
```
[the remark this answers is not in the document]
Comment (eddieh-xlnx) [reply src/com/xilinx/rapidwright/router/SATRouter.java]: Resolved by #899 which is now part of this PR.

File: test/src/com/xilinx/rapidwright/design/tools/TestLUTTools.java [hunk 2]
```diff
@@ -66,4 +74,52 @@ public void testGetCompanionLUTName() {
             Device.releaseDeviceReferences();
         }
     }
+
+    @Test
+    public void testFixPinSwaps() {
```
Comment (eddieh-xlnx) [inline test/src/com/xilinx/rapidwright/design/tools/TestLUTTools.java:79]:
```suggestion
    public void testUpdateLutPinSwapsFromPIPs() {
```

**Pull request merged**